tmk_core/protocol/ps2_mouse.c

```diff
@@ -150,6 +150,13 @@ static inline void ps2_mouse_convert_report_to_hid(report_mouse_t *mouse_report)
     // remove sign and overflow flags
     mouse_report->buttons &= PS2_MOUSE_BTN_MASK;
 
+#ifdef PS2_MOUSE_INVERT_BUTTONS
+    int buttons = mouse_report->buttons;
+    buttons |= ((mouse_report->buttons >> PS2_MOUSE_BTN_LEFT) & 1U) << 1;
+    buttons |= (mouse_report->buttons >> PS2_MOUSE_BTN_RIGHT) & 1U;
+    mouse_report->buttons = buttons;
```

Comment:
Not a fan of the `PS2_MOUSE_BTN_RIGHT == PS2_MOUSE_BTN_LEFT << 1` assumption here. I haven't tested but i think it should be more like:
```suggestion
    mouse_report->buttons ^= 1 << PS2_MOUSE_BTN_LEFT; 
    mouse_report->buttons ^= 1 << PS2_MOUSE_BTN_RIGHT; 
```

Reply:
I tested your code and doesn't work as expected.

What about this?
```
#ifdef PS2_MOUSE_INVERT_BUTTONS
    mouse_report->buttons = (mouse_report->buttons >> PS2_MOUSE_BTN_LEFT & 1U) << PS2_MOUSE_BTN_RIGHT | (mouse_report->buttons >> PS2_MOUSE_BTN_RIGHT & 1U) << PS2_MOUSE_BTN_LEFT | (mouse_report->buttons >> PS2_MOUSE_BTN_MIDDLE & 1U) << PS2_MOUSE_BTN_MIDDLE;
#endif
```
I tested with all the buttons ans works well.

Reply:
If you prefer something a more agnostic to constants maybe this works.
```
#ifdef PS2_MOUSE_INVERT_BUTTONS
    mouse_report->buttons = \
        // Change left to rigth
        (mouse_report->buttons >> PS2_MOUSE_BTN_LEFT & 1U) << PS2_MOUSE_BTN_RIGHT | \
        // Change rigth to left
        (mouse_report->buttons >> PS2_MOUSE_BTN_RIGHT & 1U) << PS2_MOUSE_BTN_LEFT | \
        // Leave the rest untoched
        (mouse_report->buttons & ~(
            (PS2_MOUSE_BTN_MASK >> PS2_MOUSE_BTN_LEFT & 1U) << PS2_MOUSE_BTN_LEFT | \
            (PS2_MOUSE_BTN_MASK >> PS2_MOUSE_BTN_RIGHT & 1U) << PS2_MOUSE_BTN_RIGHT \
        ));
#endif
```